### PR TITLE
Drop xrootd from alien-runtime dependencies

### PR DESCRIPTION
**Pull request by ktf**
As discussed with @Atlantic777, we need to drop those in order to be able to break the circular dependency when building xrootd python bindings.

### CODE REVIEW

**Comment by Atlantic777** [comment]
@ktf it looks like we actually want to keep ApMon-CPP in AliEn-Runtime if possible.

File: root.sh [hunk 1]
```diff
@@ -225,10 +225,10 @@ module load BASE/1.0 ${ALIEN_RUNTIME_VERSION:+AliEn-Runtime/$ALIEN_RUNTIME_VERSI
                      ${LZMA_VERSION:+lzma/$LZMA_VERSION-$LZMA_REVISION}
 # Our environment
 setenv ROOT_RELEASE \$version
-setenv ROOT_BASEDIR \$::env(BASEDIR)/$PKGNAME
-setenv ROOTSYS \$::env(ROOT_BASEDIR)/\$::env(ROOT_RELEASE)
-prepend-path PYTHONPATH \$::env(ROOTSYS)/lib
-prepend-path PATH \$::env(ROOTSYS)/bin
-prepend-path LD_LIBRARY_PATH \$::env(ROOTSYS)/lib
+setenv ROOT_BASEDIR \$BASEDIR/$PKGNAME
```
Comment (ktf) [inline root.sh]: I think here you do need $::env

Comment (ktf) [reply root.sh]: It's only prepend-path which cannot have `:`

File: xalienfs.sh [hunk 2]
```diff
@@ -61,7 +64,9 @@ module load BASE/1.0 XRootD/${XROOTD_VERSION}-${XROOTD_REVISION}             \\
             AliEn-Runtime/${ALIEN_RUNTIME_VERSION}-${ALIEN_RUNTIME_REVISION}
 
 # Our environment
-setenv XALIENFS_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
-prepend-path LD_LIBRARY_PATH \$::env(XALIENFS_ROOT)/lib
-prepend-path PATH \$::env(XALIENFS_ROOT)/bin
+set XALIENFS_ROOT \$BASEDIR/$PKGNAME/\$version
```
Comment (ktf) [inline xalienfs.sh]: You need `::env`

File: alien-root-legacy.sh [hunk 3]
```diff
@@ -77,12 +64,10 @@ module load BASE/1.0 ${GCC_TOOLCHAIN_VERSION:+GCC-Toolchain/$GCC_TOOLCHAIN_VERSI
...
 
 # Our environment
-setenv ALIEN_ROOT_LEGACY_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
-setenv GSHELL_ROOT \$::env(ALIEN_ROOT_LEGACY_ROOT)
-setenv GSHELL_NO_GCC 1
-prepend-path PATH \$::env(ALIEN_ROOT_LEGACY_ROOT)/bin
-prepend-path LD_LIBRARY_PATH \$::env(ALIEN_ROOT_LEGACY_ROOT)/lib
-append-path ROOT_PLUGIN_PATH \$::env(ALIEN_ROOT_LEGACY_ROOT)/etc/plugins
-prepend-path ROOT_INCLUDE_PATH \$::env(ALIEN_ROOT_LEGACY_ROOT)/include
+set ALIEN_ROOT_LEGACY_ROOT \$BASEDIR/$PKGNAME/\$version
```
Comment (ktf) [inline alien-root-legacy.sh]: You need `::env`

**Comment by Atlantic777** [comment]
@ktf so, this PR is good to go as far as I am concerned.

File: root.sh [hunk 4]
```diff
@@ -227,8 +227,10 @@ module load BASE/1.0 ${ALIEN_RUNTIME_VERSION:+AliEn-Runtime/$ALIEN_RUNTIME_VERSI
 setenv ROOT_RELEASE \$version
 setenv ROOT_BASEDIR \$::env(BASEDIR)/$PKGNAME
 setenv ROOTSYS \$::env(ROOT_BASEDIR)/\$::env(ROOT_RELEASE)
-prepend-path PYTHONPATH \$::env(ROOTSYS)/lib
-prepend-path PATH \$::env(ROOTSYS)/bin
-prepend-path LD_LIBRARY_PATH \$::env(ROOTSYS)/lib
+
+set ROOTSYS  \$::env(ROOTSYS)
+prepend-path PYTHONPATH \$ROOTSYS/lib
```
Comment (ktf) [inline root.sh]: Please let's try to keep it uniform with the rest, why not using `\$ROOT_ROOT`?

Comment (Atlantic777) [reply root.sh]: Well, as ROOTSYS was an env variable I just created a copy of it that is a normal variable. I could also rename this "normal ROOTSYS" variable to ROOT_ROOT. I'm on it.

**Pull request merged**